packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -29,6 +29,15 @@ export 'semantics_event.dart' show SemanticsEvent;
 /// Used by [SemanticsNode.visitChildren].
 typedef SemanticsNodeVisitor = bool Function(SemanticsNode node);
 
+/// Signature for the [SemanticsConfiguration.merger].
+///
+/// The input map can be used for looking up [SemanticsConfiguration]s rendering
+/// children produces. The keys are [RenderObject]s, and the values are their
```

Comment:
If the keys are always RenderObjects, I would type this as `Map<RenderObject, List<SemanticsConfiguration>>`, unless you don't want folks to access the render object, in which case I would use `Map<Object, List<SemanticsConfiguration>>`

We've definitely talked about removing the AbstractNode usage before, so I would avoid exposing it via public API.

Reply:
This is in semantics layer so I don't have access to renderobject class that is why I used AbstractNode, I will use `Object` instead

Reply:
Elsewhere we use SemanticsTag to identify SemanticsNodes coming from a particular RenderObject because RenderObjects are not a concept the semantics layer knows about. I wonder if we could reuse that concept here for the configurations as well instead of awkwardly mapping the configurations to an `Object` of unknown type because the type we really want here doesn't exist in this layer.